### PR TITLE
Add compatibility with Tesla/ Kosmos Toolbox

### PR DESCRIPTION
**Pull request by Borrajo**
😊😊
I hope this improvement is very helpful!

Team Racoon

### CODE REVIEW

**Comment by ndeadly** [comment]
Can you explain the purpose of this? If it's to be able to start/stop the module, I don't think it can be done in this case. I rely on mitm'ing the bluetooth module to steal event/shared memory handles from the system on boot.

**Comment by Borrajo** [comment]
The module is disabled by default, and it is necessary create the flags / boot2.flag file, in this way it would not be necessary. 
 On the other hand, some people use a lot of modules and that collapses the shared memory and sometimes it is useful to disable some.   I'm not saying it's a super improvement, but it can be something interesting!  
It also helps the end user to see a list of the modules activated in their system (using Tesla or Kosmos Toolbox).

**Comment by ndeadly** [comment]
Yeah, I understand why people might want to toggle sysmodules on/off, just wasn't sure if this was the sole reasoning behind wanting support for Tesla/Kosmos Toolbox. What you describe will not work for MissionControl due to the way it sits between the bluetooth module and the rest of the system. If it's not running on boot, then there is no way to acquire the handles required to intercept bluetooth events. If you boot the system with it enabled, but then kill it off, there is no way to give the original handles back to the system so Bluetooth will stop responding. Start it up again and you don't have the original handles anymore either.

**Comment by ndeadly** [comment]
The best you could do is having some way to automate swapping between MissionControl's modified `boot2` module and the original one from Atmosphere. You would still have to reboot the console to switch, but at least it could be done without messing about manually with files on the SD

**Comment by ndeadly** [comment]
Actually, re-reading your comment maybe I've misunderstood. Is this only toggling the sysmodules for next boot? If that's the case then maybe it could work with the method I described above. MissionControl doesn't currently work with the `boot2.flag` because the current mechanism in Atmosphere doesn't launch the module early enough to intercept the bluetooth initialisation. This is why the modified `boot2` is distributed. SciresM has said he will address this in the future. When this is the case it should be able to launch with the flag like any other sysmodule.

**Comment by Borrajo** [comment]
thanks for the explanation. Sorry for my poor english.

**Comment by Borrajo** [comment]
now I see that the boot2.flag file is being generated. I understand that this file is to enable or disable the mitm. Wouldn't it make sense to check this PR again? thank you very much!

**Comment by ndeadly** [comment]
Yeah, with the custom boot2 gone the sysmodule is now launched based on the boot2.flag. I'm not familiar with how kosmos toolbox works, can it be used with sysmodules that can only be enabled/disabled at boot time?

**Comment by impeeza** [comment]
Hi, putting a json file on the content folder you can instruct to kosmos/tesla(sysmodules overlay) to enable/disable by means of flag file creation and if module requiere restart for disabling

**Comment by Borrajo** [comment]
Kosmos toolbox or Tesla (sysmodules) create or delete the file boot2.flag does nothing else.

[MissionControl in Tesla](https://ibb.co/4dxRMJm)
[PS4 pad connected](https://ibb.co/yQGKsRG)
[Activate/Deactivate in Kosmos Toolbox](https://ibb.co/cFNrc4x)
[Confirmation "require restart"](https://ibb.co/ZLRt2R6)
[MissionControl OFF in Tesla](https://ibb.co/0YwR0yS)

**Comment by ndeadly** [comment]
@Borrajo if you've confirmed everything working as expected I'll merge this

**Review by ndeadly — CHANGES_REQUESTED** [review]
BTDRVMITM_TID -> BLUETOOTH_MITM_TID

Could you move this line down so that it's separate from the core sysmodule stuff? toolbox.json should also go in the bluetooth-mitm folder

**Comment by Borrajo** [comment]
Sorry, I wanted to give a quick answer, and I made a folder error

**Comment by ndeadly** [comment]
> Sorry, I wanted to give a quick answer, and I made a folder error

It's no problem, but I will merge directly into master so the code should be correct :)

**Review by ndeadly — CHANGES_REQUESTED** [review]
toolbox.json is still in the root MissionControl source directory. It should be moved to `bluetooth-mitm` and the Makefile updated accordingly

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ dist: all
 	mkdir -p dist/atmosphere/contents/$(BLUETOOTH_MITM_TID)/flags
 	touch dist/atmosphere/contents/$(BLUETOOTH_MITM_TID)/flags/boot2.flag
 	
-  	cp toolbox.json dist/atmosphere/contents/$(BLUETOOTH_MITM_TID)/toolbox.json
+  	cp $(BLUETOOTH_MITM_TID)/toolbox.json dist/atmosphere/contents/$(BLUETOOTH_MITM_TID)/toolbox.json
```
Comment (ndeadly) [inline Makefile:27]: `$(BLUETOOTH_MITM_TID)` -> `bluetooth-mitm`

**Review by ndeadly — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (Borrajo) [reply Makefile:27]: very well, I thought that this way it would be more scalable.

Comment (ndeadly) [reply Makefile]: This folder doesn't exist though. The title ID is only used for the directory on the switch side. You're copying toolbox.json from the bluetooth-mitm folder you just moved it to

**Review by ndeadly — APPROVED** [review]
Looks good. If this builds the dist zip correctly now I'll merge

**Comment by Borrajo** [comment]
everything went well? ☺

**Comment by ndeadly** [comment]
@Borrajo sorry, I haven't had a chance to verify it yet. I don't expect any issues though.

**Pull request merged**